metadata/pkg/result-paginator/ResultPaginator.go

```diff
@@ -0,0 +1,23 @@
+package result_paginator
```

Comment:
Please avoid adding new file.
1. Please don't create new file inside pkg folder(especially for query related functions). Better create a folder and then push all file inside that folder

Reply:
Can't find a more appropriate class to move the function into